packages/web-frontend/cypress/scripts/generateConfig/VisualisationFilter.js

```diff
@@ -21,6 +22,13 @@ class VisualisationFilter {
     throw new Error('Any subclass of VisualisationFilter must implement the "fetchRecords" method');
   };
 
+  /**
+   * @abstract
+   */
+  getKey = () => {
+    throw new Error('Any subclass of VisualisationFilter must implement the "getKey" method');
+  };
+
```

Comment:
Thanks for adding this method! I would like your opinion on the pattern used - I am still undecided about the best way to handle subclass properties expected by a parent class in JS. I guess that the "safer"/"more proper" OOP approach would be to define abstract getters for them in the parent class (the pattern used above).

However, in the JS world perhaps just defining those properties in the parent class with a comment (eg [DatabaseType.js#L8](https://github.com/beyondessential/tupaia/blob/dev/packages/database/src/DatabaseType.js#L8)) would be enough. Even more so for a simple case like here, with all classes defined in the same file!

If we want to make sure that devs will not forget to override those, we could use [Node assertions](https://nodejs.org/api/assert.html#assert_assert_notequal_actual_expected_message) - we haven't use those much in our codebase but they seem cool!

```js
import assert from 'assert';

class VisualisationFilter {
  // Must override in subclasses
  key = null;

  constructor() {
    assert.notEqual(this.key, null);
  }
}
```

Let me know if you have an opinion on this - they more I look into it, the more keen I am on refactoring it 😆

Reply:
That's a good point! I also second the latter! It is simpler and clearer to me because I think expressing `this.key` is always better than `this.getKey()` (if we can, and also avoid the need to assign `getKey()` to a variable). Node assertion is handy too!

Reply:
In that case, would you be keen to refactor the code to use the simpler pattern? We should also verify that node assertions in the constructor work as intended 😁

Reply:
Very happy to! I'll apply the same for other properties as well